docs/workflow-inputs.md

```diff
@@ -0,0 +1,113 @@
+# Workflow Inputs
+
+![GA](assets/ga.svg)
+
+## Introduction
+
+`Workflows` and `template`s operate on a set of defined parameters and arguments that are supplied to the running container. The precise details of how to manage the inputs can be confusing; this article attempts to clarify concepts and provide simple working examples to illustrate the various configuration options.
+
+The examples below are limited to `DAGTemplate`s and mainly focused on `parameters`, but similar reasoning applies to the other types of `template`s.
+
+### Parameter Inputs
+
+First, some clarification of terms is needed. For a glossary reference, see [Argo Core Concepts](https://argoproj.github.io/argo/core-concepts/).
+
+A `Workflow` accepts `arguments`. A `template` accepts `inputs` (`arguments` is deprecated in `template`s). The structure of both is identical.
```

Comment:
This is not quite true. `arguments` are where we pass in _live_ parameters, and `inputs` are were we _define_ those paramters. The reason why `template` accepts inputs, it's because it's a template definition. The reason why a Workflow accepts `arguments` is because the workflow is an invocation of itself.

Reply:
I'm not quite following; what part of this statement is false? Are you suggesting that I need to clarify the distinction between `arguments` vs `inputs`?

[WorkflowSpec](https://github.com/argoproj/argo/blob/master/docs/fields.md#workflowspec).[arguments](https://github.com/argoproj/argo/blob/master/docs/fields.md#arguments) is an Array of `artifacts` and `parameters`. This is identical to [Template](https://github.com/argoproj/argo/blob/master/docs/fields.md#template).[inputs](https://github.com/argoproj/argo/blob/master/docs/fields.md#inputs). Even if their purpose is different, the structure is the same.

Reply:
```suggestion
A `workflow` provides `arguments`, which are passed in to the entry point template. A `template` defines `inputs` which are then provided by template callers (such as `steps`, `dag`, or even a `workflow`). The structure of both is identical.
```

Reply:
Here is my suggestion

Reply:
Thanks, updated in latest commit.